Add documentation tests (#9)
SmartMonkey-git/robinson-group-python-template: README.md

To get ruff automatically running install the following Plugin:
- [VSCode - Ruff Plugin](https://marketplace.visualstudio.com/items?itemName=charliermarsh.ruff)

## Testing

> A test suite of a typical Python package consists of unit, integration, and documentation tests
> (i.e. the code snippets embedded in the documentation),
> and it is truly important to write the tests to ensure that the code works as intended.
> 
> The package template provides a minimal setup and the tests can be executed by invoking the Pytest runner:

Reviewer: Just removed some words and added different punctuation. No hard feelings about it.

Maybe add a link to some article that explains what unit, integration and documentation tests are. I imagine some people might not know the difference. :) 

```suggestion
A test suite of a typical Python package consists of unit, integration, and documentation tests
(i.e. the code snippets embedded in the documentation).
It is important to write tests to ensure that your code works as intended.

This template provides a minimal setup for testing. Tests can be executed by invoking Pytest:
```

Author: Looks good, thank you!

```shell
pytest
```

The runner looks for tests in `src` and `tests` folders. On top of that, the Python code
in the top-level `README.md` file is also executed, so we can demonstrate
the most imporant functionality:

```python
>>> from project_name.foo import foo
>>> foo()
True

> ```

Reviewer: The way you phrased it, it sounds like the code snipped is imporant. However, I think what you actually wanted to make is an example on how to write a doctest, right?

So, maybe:

```suggestion
the most imporant functionality. For example snippets like these will be executed by pytest:

```python
>>> from project_name.foo import foo
>>> foo()
True

```
```

> Since the documentation is executed, any code changes that can render the documentation obsolete can be picked up,
> prompting the documentation update.

Reviewer: The diff looks weird here, but only changed the word can to will.
```suggestion
Since the documentation is executed, any code changes that render the documentation obsolete will be picked up, forcing a documentation update.
```

Author: Thank you!
